pipeline/batch.go

```diff
@@ -118,19 +165,97 @@ func (b *Batcher) Start(ctx context.Context) {
 		})
 	}
 
+	// delete old committed counters
+	go b.invalidateOldCommittedCounters(ctx, keepBatcherCommitsCount)
+
+	batcherTimeChan := make(chan BatcherTimeDTO, b.opts.Workers)
+	b.batcherTimeChan = batcherTimeChan
+	go b.updateCommitWaitValues(ctx, batcherTimeChan)
+
 	longpanic.Go(b.heartbeat)
 }
 
-type WorkerData any
+// invalidateOldCommittedCounters removes old committed counters
+func (b *Batcher) invalidateOldCommittedCounters(ctx context.Context, seconds int64) {
+	for {
+		t := time.NewTicker(time.Second * 30)
+		select {
+		case <-ctx.Done():
+			return
+		case <-t.C:
+			now := time.Now()
+			b.committedCountersMu.Lock()
+			for k := range b.committedCounters {
+				if now.Unix()-k > seconds {
+					delete(b.committedCounters, k)
+				}
+			}
+			b.committedCountersMu.Unlock()
+		}
+	}
+}
+
+func (b *Batcher) updateCommitWaitValues(ctx context.Context, batcherTimeChan <-chan BatcherTimeDTO) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case dto := <-batcherTimeChan:
```

Comment:
why are we not immediately updating batcherTimeWaitMap in saveObservabilityInfo? Thus, the separate goroutine will not be needed

Reply:
done.